tests/Database/DatabaseEloquentBelongsToManyTest.php

```diff
@@ -116,7 +116,7 @@ public function testModelsAreProperlyMatchedToParents()
         $this->assertEquals(2, $models[1]->foo[0]->pivot->user_id);
         $this->assertEquals(2, $models[1]->foo[1]->pivot->user_id);
         $this->assertEquals(2, count($models[1]->foo));
-        $this->assertEquals(0, count($models[2]->foo));
+        $this->assertEquals(0, count((array) $models[2]->foo));
```

Comment:
If we are going to change it in 5.5 - then the current test's should be left unchanged? That way we can prove the problem is solved?

Because *technically* we are now saying that 5.4 is not really suitable for PHP 7.2? You have situations where the current code will fail - so hiding those results in a changed test will trip people up?

Plus - this is a change that even people with their own test suites might miss during an upgrade. I doubt people are having full application test suites that always specifically tests for a `null` return (i.e. a `0` count) result as part of the test. So what will occur is people start upgrading, think they have "all green" - and when it hits production start getting errors.

Whilst it's not Laravel's fault - the reality is 5.4 is probably only suitable up to PHP 7.1, and 5.5 is suitable for 7.2 on onwards? ¯\_(ツ)_/¯